src/System.IO.Pipelines/UnownedBufferReader.cs

```diff
@@ -194,12 +195,13 @@ void IPipelineReader.Advance(ReadCursor consumed, ReadCursor examined)
             }
 
             // Again, we don't need an interlock here because Read and Write proceed serially.
-            if (!examined.IsDefault &&
+            if ((!examined.IsDefault &&
```

Comment:
There seems to be quite a bit of duplication between PipelineReaderWriter and UnownedBufferReader.